### PR TITLE
Add fine grain execution control for pyvista-plot directive

### PR DESCRIPTION
**Pull request by bjlittle**
### Overview

This pull-request adds some fine grain options to control the execution of the `pyvista-plot` directive.

#### Motivation

Building documentation can be expensive, particularly if it's image rich, which is pretty much the accepted norm here. However, sometimes as a developer there is a need for a slicker workflow experience when changing the documentation i.e., change the docs, build the docs, review the rendered docs ... typically this can involve several rinse'n repeat iterations, but the major bottleneck is always the documentation build time.

I'm proposing a couple of minor changes that will hopefully give developers the ability to improve the turnaround of their documentation changes by rendering only images of their choosing, which may be none, all or some.

##### Directive Options

The `pyvista-plot` directive has the following **new** options:

- `:skip:` which has an optional boolean argument i.e., `yes`, `1`, `true`, `no`, `0`, `false`. If no argument is provided, then it is assumed to be enabled e.g., `:skip:` is equivalent to `:skip: yes`. The purpose of this option is to override the behaviour of the `plot_skip` configuration variable, and enforce the execution of the directive or not
- `:optional:` which is a flag (no argument). The purpose of this option is to mark a directive for conditional execution, which is controlled by the state of the `plot_skip_optional` configuration variable.

##### Configuration Variables (`conf.py`)

The `pyvista-plot` directive has the following **new** configuration variables:

- `plot_skip` is a boolean. Default is `False`. This variable is used to seed the value of any directive that doesn't explicitly specify a `:skip:` option
- `plot_skip_optional` is a boolean. Default is `False`. This variable is used to control whether any directive marked with an `:optional:` flag is executed or not

#### Workflow

Note that, given the default values of the **new** `conf.py` configuration variables there is no change in the existing behaviour of the `pyvista-plot` directive.

Here are some example workflows that are now easily possible:

- disable all `pyvista-plot` directives with `plot_skip = True` in the `conf.py`
- disable all `pyvista-plot` directives with `plot_skip = True` apart from one (or more) marked with `:skip: no`
- execute all `pyvista-plot` directives with `plot_skip = False` and `plot_skip_optional = False` (default behaviour)
- execute all `pyvista-plot` directives with `plot_skip = False` apart from one (or more) marked with `:optional:` because `plot_skip_optional = True`

Note that, skipped `pyvista-plot` directives will still have their code blocks rendered but with no image and no caption.

### CODE REVIEW

**Comment by bjlittle** [comment]
pre-commit.ci autofix

File: pyvista/ext/plot_directive.py [hunk 1]
```diff
@@ -55,7 +55,7 @@
         directives for which the ``:context:`` option was specified.  This only
         applies to inline code plot directives, not those run from files.
 
-    nofigs : bool
+    nofigs : None
```
Comment (tkoyama010) [inline pyvista/ext/plot_directive.py:58]: ref https://numpydoc.readthedocs.io/en/stable/format.html#parameters
```suggestion
    nofigs : bool, default None
```

File: pyvista/ext/plot_directive.py [hunk 2]
```diff
@@ -64,10 +64,22 @@
         figure. This overwrites the caption given in the content, when the plot
         is generated from a file.
 
-    force_static : bool
+    force_static : None
```
Comment (tkoyama010) [inline pyvista/ext/plot_directive.py:67]:
```suggestion
    force_static : bool, default None
```

File: pyvista/ext/plot_directive.py [hunk 3]
```diff
@@ -95,6 +108,14 @@
     plot_cleanup : str
         Python code to be run after every plot directive block.
 
+    plot_skip : bool
+        Default value for the ``skip`` directive option.  Default is
+        ``False``.
```
Comment (tkoyama010) [inline pyvista/ext/plot_directive.py:113]:
```suggestion
    plot_skip : bool, default False
        Default value for the ``skip`` directive option.
```

File: pyvista/ext/plot_directive.py [hunk 4]
```diff
@@ -95,6 +108,14 @@
...
+
+    plot_skip_optional : bool
+        Whether to skip execution of ``optional`` directives.  Default is
+        ``False``.
+
```
Comment (tkoyama010) [inline pyvista/ext/plot_directive.py:118]:
```suggestion

    plot_skip_optional : bool, default False
        Whether to skip execution of ``optional`` directives.

```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Thank you for the change. It would be grateful if you could add a test to the `tests/test_tinypages.py` so that we can understand the specific use of this change.

**Comment by bjlittle** [comment]
@tkoyama010 `uber/h3-py` have just release a breaking major 4.x release ... patching `geovista` now :+1:

**Comment by bjlittle** [comment]
pre-commit.ci autofix

**Comment by bjlittle** [comment]
@tkoyama010 Back to you :+1: 

I've updated the test coverage as you requested. Hopefully this helps.

If you have any further questions then please don't hesitate to ask.

Also, I appear to now have a failing `Vale` CI tests ... not quite sure how to fix that. Can you advise please :pray:

**Comment by bjlittle** [comment]
Ahhh the same `vale` barf appears to be happening with other pull-requests e.g., #6707

So there seems to be a systemic failure here, but since I don't use and understand `vale` not quite sure what to do ...

**Comment by tkoyama010** [comment]
> Ahhh the same `vale` barf appears to be happening with other pull-requests e.g., #6707
> 
> So there seems to be a systemic failure here, but since I don't use and understand `vale` not quite sure what to do ...

It is resolved in #6708 .

**Comment by bjlittle** [comment]
@tkoyama010 Okay, back to you as the `vale` CI job seems to be behaving itself again.

Thanks :+1:

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Comment by bjlittle** [comment]
@tkoyama010 Awesome! Thanks! Much appreciated :beers: :partying_face: :tada:

**Pull request merged**